lib/event_sourcery/event_processing/esp_runner.rb

```diff
@@ -61,12 +77,12 @@ def listen_for_shutdown_signals
         end
       end
 
-      def shutdown
-        @shutdown_requested = true
-      end
-
-      def wait_till_shutdown_requested
-        sleep(1) until @shutdown_requested
+      def while_waiting_for_shutdown
+        yield
+        until shutdown_requested?
+          sleep(1)
+          yield
+        end
```

Comment:
So `record_terminated_processes` needs calling twice to be effective? Curious why the double call is needed

Reply:
It's not that it needs to be called extra times, it's the order I'm more interested. Perhaps this would do a better job of revealing the intent:
```ruby
loop do
  record_terminated_processes
  break if shutdown_requested?
  sleep(1)
end
```
I'd like to guarantee `record_terminated_processes` gets called at least before loop ends on the break condition.

Reply:
Yeah I think that would be better

Reply:
Updated in fb14c10e